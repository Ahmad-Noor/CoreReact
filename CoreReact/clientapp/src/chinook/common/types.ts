export interface IEmployee{
    employeeId:	number
    lastName: string
    firstName: string
    title: string
    reportsTo: number
    birthDate: string
    hireDate: string
    address: string
    city: string
    state: string
    country: string
    postalCode: string
    phone: string
    fax: string
    email: string
}

export interface IAlbum{
    albumId: number
    title:	string
    artistId: number
    artist: IArtist
    tracks:	ITrack[]
}

export interface IArtist{
    artistId: number
    name: string
    albums: IAlbum[]
}

export interface ITrack{
    trackId: number
    name: string
    albumId: number
    mediaTypeId: number
    genreId: number
    composer: string
    milliseconds: number
    bytes: number
    unitPrice:	string
}

export interface IGenre{
    genreId: number
    name:string
    tracks:ITrack[]
}

export interface IChinookStates{
    employeesState: IEmployee[]
    albumsState: IAlbum[]
    artistsState: IArtist[]
    genresState: IGenre[]
    tracksState: ITrack[]
    customersState: ICustomer[]
    invoicesState: IInvoice[]
    saleRecordsState: ISaleRecord[]
    shoppingCart: IShoppingCart
    loggedInEmployeeId: number
    selectedCustomerId: number
    viewState: ChinookView
}

export interface ICustomer{
    customerId: number
    firstName: string
    lastName: string
    company: string
    address: string
    city: string
    state: string
    country: string
    postalCode: string
    phone: string
    fax: string
    email: string
    supportRepId: number
}

export interface IShoppingCart{
    trackIds: number[]
    albumIds: number[]
}

export interface IInvoice{
    invoiceId: number
    customerId: number
    invoiceDate: string
    billingAddress: string
    billingCity: string
    billingState: string
    billingCountry:	string
    billingPostalCode: string
    total: string
    invoiceItems: IInvoiceItem[]
}

export interface IInvoiceItem{
    invoiceLineId: number
    invoiceId: number
    trackId: number
    unitPrice: string
}

export type ChinookView = 'store' | 'checkout' | 'report'

export interface IOrder{
    employeeId: number
    customerId: number
    trackIds: number[]
    totalPrice: number    
}

export interface ISaleRecord{
    saleRecordId: number
    employeeId: number
    invoiceId: number   
}
