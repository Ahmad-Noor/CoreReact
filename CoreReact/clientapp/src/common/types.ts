import { INorthwindStates } from "../northwind/common/types";
import { IChinookStates } from "../chinook/common/types";

export interface IHomeState{
    pageLoaded: boolean
} 

export interface IStore{
    homeState: IHomeState
    newsState: INewsState
    northwindStates: INorthwindStates
    chinookStates: IChinookStates
}

export interface INewsSource
{
    id: string
    name: string
}

export interface INews{
    source: INewsSource
    author: string
    title: string
    description: string
    url: string
    urlToImage: string
    publishedAt: Date
}

export interface INewsState{
    loading: boolean
    news: INews[]
}
