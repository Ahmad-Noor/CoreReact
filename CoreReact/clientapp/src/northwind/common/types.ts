export interface IEmployee{
    id: number
    lastName: string
    firstName: string
    title: string
    titleOfCourtesy: string
    birthDate: string
    hireDate: string
    address: string
    city: string
    region: string
    postalCode: string
    country: string
    homePhone: string
    extension: string
    photo: string
    notes: string
    reportsTo: number
    photoPath: string    
}

export type EmployeeViewType = 'new' | 'edit' | 'list'
export interface IEmployeeViewState{
    type: EmployeeViewType
    id: number
    employee?: IEmployee
}

export interface IEmployeeApi {
    url: string
    method: string
    data: IEmployee
}

export interface INorthwindTabState{
    eventKey: string
}

export interface INorthwindStates{
    northwindTabState: INorthwindTabState
    employeeViewState: IEmployeeViewState
    customerViewState: ICustomerViewState
    orderViewState: IOrderViewState
    orderDetailState: IOrderDetail[]
}

export interface ICustomerViewState{
    id: string
}

export interface ICustomer{
    id:	string
    companyName: string
    contactName: string
    contactTitle: string
    address: string
    city: string
    region: string
    postalCode: string
    country: string
    phone: string
    fax: string
}

export interface IOrder {
    id: number
    customerId: string
    employeeId: number
    orderDate: string
    requiredDate: string
    shippedDate: string
    shipVia: number
    freight: string
    shipName: string
    shipAddress: string
    shipCity: string
    shipRegion: string
    shipPostalCode: string
    shipCountry: string
    customer: ICustomer
}

export interface IOrderViewState{
    loading: boolean
    orders: IOrder[]
}

export interface IOrderDetail{
    id: string
    orderId: number
    productId: number
    unitPrice: string
    quantity: number
    discount: number
    product: IProduct
}

export interface IProduct{
    id: number
    productName: string
    supplierId: number
    categoryId: number
    quantityPerUnit: string
    unitPrice: string
    unitsInStock: number
    unitsOnOrder: number
    reorderLevel: number
    discontinued: number
    supplier: ISupplier
  }

  export interface ISupplier{
    id: number
    companyName: string
    contactName: string
    contactTitle: string
    address: string
    city: string
    region: string
    postalCode: string
    country: string
    phone: string
    fax: string
    homePage: string
  }
