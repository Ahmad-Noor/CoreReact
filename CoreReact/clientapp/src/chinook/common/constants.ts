export const ALBUMS_LOADED = 'ALBUMS_LOADED'
export const GENRES_LOADED = 'GENRES_LOADED'
export const ARTISTS_LOADED = 'ARTISTS_LOADED' 
export const TRACKS_LOADED = 'TRACKS_LOADED'
export const EMPLOYEES_LOADED = 'EMPLOYEES_LOADED'
export const CUSTOMERS_LOADED = 'CUSTOMERS_LOADED'
export const INVOICES_LOADED = 'INVOICES_LOADED'
export const SALERECORD_LOADED = 'SALERECORD_LOADED'

export const EMPLOYEE_LOGGED_IN = 'EMPLOYEE_LOGGED_IN'
export const EMPLOYEE_LOGGED_OUT = 'EMPLOYEE_LOGGED_OUT'

export const ADD_TRACK = 'ADD_TRACK'
export const REMOVE_TRACK = 'REMOVE_TRACK'
export const ADD_ALBUM = 'ADD_ALBUM'
export const REMOVE_ALBUM = 'REMOVE_ALBUM'
export const CLEAR_SHOPPINGCART = 'CLEAR_SHOPPINGCART'

export const CUSTOMER_SELECTED = 'CUSTOMER_SELECTED'
export const CHANGE_VIEW = 'CHANGE_VIEW'