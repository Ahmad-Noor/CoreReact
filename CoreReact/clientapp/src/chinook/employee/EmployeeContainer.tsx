import * as React from 'react'

export class EmployeeContainer extends React.Component {
    render(){
        return (
            <h2>EmployeeContainer</h2>
        )
    }
}