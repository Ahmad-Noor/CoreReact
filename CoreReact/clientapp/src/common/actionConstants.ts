export const HOMEPAGE_LOADED = 'HOMEPAGE_LOADED'

export const GET_NEWS_STARTING = 'GET_NEWS_STARTING'
export const GET_NEWS_COMPELED = 'GET_NEWS_COMPELED'

