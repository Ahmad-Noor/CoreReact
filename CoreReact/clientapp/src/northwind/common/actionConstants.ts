export const ENTER_EMPLOYEE_EDITING = 'ENTER_EMPLOYEE_EDITING'
export const EMPLOYEE_LOADED = 'EMPLOYEE_LOADED'
export const OUTOF_EMPLOYEE_EDITING = 'OUTOF_EMPLOYEE_EDITING'

export const NORTHWIND_TAB_CHANGED = 'NORTHWIND_TAB_CHANGED'

export const ENTER_CUSTOMER_EDITING = 'ENTER_CUSTOMER_EDITING'
export const OUTOF_CUSTOMER_EDITING = 'OUT_CUSTOMER_EDITING'

export const ORDER_LOAD_STARTED = 'ORDER_LOAD_STARTED'
export const ORDER_LOAD_COMPELED = 'ORDER_LOAD_COMPELED'
export const ORDERDETAILS_LOAD_COMPELED = 'ORDERDETAILS_LOAD_COMPELED'