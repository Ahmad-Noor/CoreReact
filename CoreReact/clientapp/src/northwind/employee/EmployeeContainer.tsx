import * as React from 'react'
import { IStore } from '../../common/types'
import Employee from './Employee'
import { EmployeeList } from './EmployeeList'
import { connect } from 'react-redux'
import { IEmployeeViewState } from '../common/types'
import { getEmployees, deleteEmployee, saveEmployee } from '../common/api'

interface IStateFromProps{
    employeeViewState: IEmployeeViewState
}

const mapStateToProps = (store: IStore):IStateFromProps => {
    return {employeeViewState: store.northwindStates.employeeViewState}
}

export class EmployeeContainerComponent extends React.Component<IStateFromProps, any> {
    render() {
        const viewState = this.props.employeeViewState
        const showEmployeeEditing = viewState && viewState.type !== 'list'
        if(showEmployeeEditing){
            return (
                <Employee save={saveEmployee} />
            )
        }
        return (
            <EmployeeList getEmployees={getEmployees} deleteEmployee={deleteEmployee} />
        )
    }
}

export const EmployeeContainer = connect(mapStateToProps, null)(EmployeeContainerComponent);