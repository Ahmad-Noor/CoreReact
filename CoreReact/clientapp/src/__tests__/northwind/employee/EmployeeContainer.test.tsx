import * as React from 'react'
import { shallow } from 'enzyme'
import { EmployeeContainerComponent } from '../../../northwind/employee/EmployeeContainer'
import { IEmployeeViewState } from '../../../northwind/common/types';

it('passes snapshot tests with list view', () => {
    // Arrange
    const state = {type: 'list'} as IEmployeeViewState

    // Act
    const target = shallow(<EmployeeContainerComponent employeeViewState={state} />)

    // Assert
    expect(target).toMatchSnapshot()
})

it('passes snapshot tests with new view', () => {
    // Arrange
    const state = {type: 'new'} as IEmployeeViewState

    // Act
    const target = shallow(<EmployeeContainerComponent employeeViewState={state} />)

    // Assert
    expect(target).toMatchSnapshot()
})