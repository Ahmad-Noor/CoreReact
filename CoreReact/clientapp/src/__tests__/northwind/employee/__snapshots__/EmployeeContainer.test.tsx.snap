// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot tests with list view 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmployeeContainerComponent
    employeeViewState={
      Object {
        "type": "list",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`passes snapshot tests with new view 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmployeeContainerComponent
    employeeViewState={
      Object {
        "type": "new",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
